design_patterns_poc/src/cache/Cache.java

```diff
@@ -0,0 +1,124 @@
+package cache;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Cache<K,V> {
+    private static Cache cacheObject;
+
+    private Map<K, CacheItem> map;
```

Comment:
The Generics <K, CacheItem> is wrong over here. This `CacheItem` isn't similar to the `CacheItem.java` that is defined in the another file.